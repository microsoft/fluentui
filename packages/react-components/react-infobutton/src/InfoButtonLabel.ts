export * from './components/InfoButtonLabel/index';
