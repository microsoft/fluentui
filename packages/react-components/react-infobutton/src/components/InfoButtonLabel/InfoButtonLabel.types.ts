import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type InfoButtonLabelSlots = {
  root: Slot<'div'>;
};

/**
 * InfoButtonLabel Props
 */
export type InfoButtonLabelProps = ComponentProps<InfoButtonLabelSlots> & {};

/**
 * State used in rendering InfoButtonLabel
 */
export type InfoButtonLabelState = ComponentState<InfoButtonLabelSlots>;
// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from InfoButtonLabelProps.
// & Required<Pick<InfoButtonLabelProps, 'propName'>>
