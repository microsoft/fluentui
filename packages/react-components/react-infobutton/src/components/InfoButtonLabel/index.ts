export * from './InfoButtonLabel';
export * from './InfoButtonLabel.types';
export * from './renderInfoButtonLabel';
export * from './useInfoButtonLabel';
export * from './useInfoButtonLabelStyles.styles';
