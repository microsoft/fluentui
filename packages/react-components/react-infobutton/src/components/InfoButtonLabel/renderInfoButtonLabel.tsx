/** @jsxRuntime classic */
/** @jsx createElement */

import { createElement } from '@fluentui/react-jsx-runtime';
import { getSlotsNext } from '@fluentui/react-utilities';
import type { InfoButtonLabelState, InfoButtonLabelSlots } from './InfoButtonLabel.types';

/**
 * Render the final JSX of InfoButtonLabel
 */
export const renderInfoButtonLabel_unstable = (state: InfoButtonLabelState) => {
  const { slots, slotProps } = getSlotsNext<InfoButtonLabelSlots>(state);

  // TODO Add additional slots in the appropriate place
  return <slots.root {...slotProps.root} />;
};
