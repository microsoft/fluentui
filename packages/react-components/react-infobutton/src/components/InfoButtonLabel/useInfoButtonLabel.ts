import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { InfoButtonLabelProps, InfoButtonLabelState } from './InfoButtonLabel.types';

/**
 * Create the state required to render InfoButtonLabel.
 *
 * The returned state can be modified with hooks such as useInfoButtonLabelStyles_unstable,
 * before being passed to renderInfoButtonLabel_unstable.
 *
 * @param props - props from this instance of InfoButtonLabel
 * @param ref - reference to root HTMLElement of InfoButtonLabel
 */
export const useInfoButtonLabel_unstable = (
  props: InfoButtonLabelProps,
  ref: React.Ref<HTMLElement>,
): InfoButtonLabelState => {
  return {
    // TODO add appropriate props/defaults
    components: {
      // TODO add each slot's element type or component
      root: 'div',
    },
    // TODO add appropriate slots, for example:
    // mySlot: resolveShorthand(props.mySlot),
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),
  };
};
