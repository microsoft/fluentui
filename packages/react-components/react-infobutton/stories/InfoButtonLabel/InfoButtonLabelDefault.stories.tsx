import * as React from 'react';
import { InfoButtonLabel, InfoButtonLabelProps } from '@fluentui/react-infobutton';

export const Default = (props: Partial<InfoButtonLabelProps>) => <InfoButtonLabel {...props} />;
