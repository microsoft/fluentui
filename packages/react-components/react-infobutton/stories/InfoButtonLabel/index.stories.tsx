import { InfoButtonLabel } from '@fluentui/react-infobutton';

import descriptionMd from './InfoButtonLabelDescription.md';
import bestPracticesMd from './InfoButtonLabelBestPractices.md';

export { Default } from './InfoButtonLabelDefault.stories';

export default {
  title: 'Preview Components/InfoButtonLabel',
  component: InfoButtonLabel,
  parameters: {
    docs: {
      description: {
        component: [descriptionMd, bestPracticesMd].join('\n'),
      },
    },
  },
};
