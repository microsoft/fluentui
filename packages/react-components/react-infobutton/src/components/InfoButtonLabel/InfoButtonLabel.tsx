import * as React from 'react';
import { useInfoButtonLabel_unstable } from './useInfoButtonLabel';
import { renderInfoButtonLabel_unstable } from './renderInfoButtonLabel';
import { useInfoButtonLabelStyles_unstable } from './useInfoButtonLabelStyles.styles';
import type { InfoButtonLabelProps } from './InfoButtonLabel.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * InfoButtonLabel component - TODO: add more docs
 */
export const InfoButtonLabel: ForwardRefComponent<InfoButtonLabelProps> = React.forwardRef((props, ref) => {
  const state = useInfoButtonLabel_unstable(props, ref);

  useInfoButtonLabelStyles_unstable(state);
  return renderInfoButtonLabel_unstable(state);
});

InfoButtonLabel.displayName = 'InfoButtonLabel';
