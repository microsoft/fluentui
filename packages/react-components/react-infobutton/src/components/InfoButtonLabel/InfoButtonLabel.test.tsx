import * as React from 'react';
import { render } from '@testing-library/react';
import { InfoButtonLabel } from './InfoButtonLabel';
import { isConformant } from '../../testing/isConformant';

describe('InfoButtonLabel', () => {
  isConformant({
    Component: InfoButtonLabel,
    displayName: 'InfoButtonLabel',
  });

  // TODO add more tests here, and create visual regression tests in /apps/vr-tests

  it('renders a default state', () => {
    const result = render(<InfoButtonLabel>Default InfoButtonLabel</InfoButtonLabel>);
    expect(result.container).toMatchSnapshot();
  });
});
