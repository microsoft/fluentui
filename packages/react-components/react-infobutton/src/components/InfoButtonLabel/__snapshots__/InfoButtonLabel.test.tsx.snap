// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoButtonLabel renders a default state 1`] = `
<div>
  <div
    class="fui-InfoButtonLabel"
  >
    Default InfoButtonLabel
  </div>
</div>
`;
